src/DynamoPackages/PackageManagerExtension.cs

```diff
@@ -163,6 +165,40 @@ public void Startup(StartupParams startupParams)
             noNetworkMode = startupParams.NoNetworkMode;
         }
 
+        /// <summary>
+        /// When a new engine is added from a package, add its dependency to the respective package in the dictionary.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        internal void PythonEngineAdded(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            if (e.Action == NotifyCollectionChangedAction.Add)
+            {
+                var assem = e.NewItems[0].GetType().Assembly;
```

Comment:
Maybe add some null checks or wrap it in a try catch. Exceptions thrown in event handler ca cause havoc